src/pool.rs

```diff
@@ -177,6 +183,41 @@ impl<Conn: Connection> PoolInner<Conn> {
         new_backends
     }
 
+    // Forcefully fail the next client claim request.
+    fn fail_claim(&mut self) {
+        let Some(request) = self.request_queue.pop_front() else {
+            return;
+        };
+        // We would have claimed this request if we could have done
+        // so earlier. Don't bother trying to access it now.
+        //
+        // Instead, identify "why haven't we succeeded" to help the
+        // client diagnose what's happening.
+        let err = match self.slots.len() {
+            // We don't know of any valid backends from the resolver
+            0 => Error::NoBackends,
+            _ => {
+                if self
+                    .slots
+                    .values()
+                    .any(|set| matches!(set.get_state(), slot::SetState::Online { .. }))
+                {
+                    // Backends exist, and appear online, but we couldn't get a claim.
+                    // Presumably, this means all existing claims are in-use.
+                    Error::AllClaimsUsed
```

Comment:
What would this mean for a client - they need to bump up the number of slots they allow the pool to have?

Maybe a slightly different question / suggestion - should this error include counts of online / offline slots? At that point maybe `AllClaimsUsed` could be combined with `NoBackendsOnline` into a single `NoClaimsAvailable { claims_used: M, offline_backends: N }` or something like that?

Reply:
bumping up the slot count would be one way to cope here, another option would be to try again later.

This could happen if all claims are used for the duration of the retry, and they aren't returned to the pool (or if there are just so many outstanding requests that we don't get to this one in time).

I can expose these internal numbers about the state, but my main hesitation with specific numbers here is that they can get out-of-date extremely quickly - before this request even completes. Also, defining "claims_used" is really important -- it **could** be that all the claims are checked-out, but it also **could** be that all claims are being recycled, but they're just taking a long time to process their `is_healthy` check.

The fullest picture we could give would be from https://docs.rs/qorb/0.1.2/qorb/pool/struct.Pool.html#method.stats  - which we could return here! - but also, presumably, the caller could call this function themselves to access this specific data.

Reply:
Ah, yeah, I didn't realize the pool stats were publicly accessible. Agreed.

(Side note on "publicly accessible" - clicking through docs.rs I eventually land on https://docs.rs/qorb/0.1.2/qorb/pool/struct.BackendStats.html#method.get, which does not have a link to the `Stats` struct it returns. Maybe that's `pub` but can't be named because it's under a non-pub module somewhere?)

Reply:
ah, good feedback, I'll get a separate PR up to address / document the stats